src/Bicep.Core.Samples/DataSet.cs

```diff
@@ -68,6 +76,8 @@ public record ExternalModuleMetadata(string Target);
 
         private readonly Lazy<ImmutableDictionary<string, ExternalModuleInfo>> lazyModulesToPublish;
 
+        private readonly Lazy<ImmutableDictionary<string, ExternalProviderInfo>> lazyProvidersToPublish;
+
         private readonly Lazy<ImmutableDictionary<string, ExternalModuleInfo>> lazyTemplateSpecs;
 
         public DataSet(string name)
```

Comment:
I don't think you need the changes in `DataSet.cs` or `DataSetsExtensions.cs` any more - could you revert these changes?